### PR TITLE
Faster carrier frequency assignment

### PR DESCRIPTION
**Pull request by jtec**
By adding `@line_profiler.profile`, and running

 https://github.com/jtec/prx/blob/85c006479a402380ecac369a7feacaf6bdf8a6c0/src/prx/test/test_main.py#L139 

with `LINE_PROFILE=1`

we can see that assigning carrier frequencies to observations takes a whooping 13 % of the time spent in https://github.com/jtec/prx/blob/9a90fc804f81e29af47a296a41dfe951502b022c/src/prx/main.py#L211

```
   426         2     230039.0 115019.5     13.1      flat_obs.loc[:, "carrier_frequency_hz"] = flat_obs.apply(
   427         1          3.0      3.0      0.0          signal_2_carrier_frequency, axis=1
   428                                               )
```

This PR reduces this to 1.8%:

```
 413         2        234.0    117.0      0.0      glo_cdma = flat_obs[
   414         2       4526.0   2263.0      0.3          (flat_obs.satellite.str[0] == "R")
   415         1       4537.0   4537.0      0.3          & (flat_obs["observation_type"].str[1].astype(int) > 2)
   416                                               ]
   417         1        369.0    369.0      0.0      flat_obs.loc[glo_cdma.index, "frequency_slot"] = int(1)
   418                                           
   419         1          4.0      4.0      0.0      def assign_carrier_frequencies(flat_obs):
   420                                                   freq_dict = pd.json_normalize(carrier_frequencies_hz(), sep="_").to_dict(
   421                                                       orient="records"
   422                                                   )[0]
   423                                                   assignable = flat_obs.frequency_slot.notna()
   424                                                   keys = (
   425                                                       flat_obs.satellite[assignable].str[0]
   426                                                       + "_L"
   427                                                       + flat_obs["observation_type"][assignable].str[1]
   428                                                       + "_"
   429                                                       + flat_obs.frequency_slot[assignable].astype(int).astype(str)
   430                                                   )
   431                                                   flat_obs.loc[:, "carrier_frequency_hz"] = keys.map(freq_dict)
   432                                                   return flat_obs
   433                                           
   434         1      18716.0  18716.0      1.2      flat_obs = assign_carrier_frequencies(flat_obs)

```

### CODE REVIEW

File: src/prx/constants.py [hunk 1]
```diff
@@ -66,8 +66,9 @@ def carrier_frequencies_hz():
         cf["R"]["L1"][frequency_slot] = (1602 + frequency_slot * 9 / 16) * cHzPerMhz
         cf["R"]["L2"][frequency_slot] = (1246 + frequency_slot * 7 / 16) * cHzPerMhz
     # Glonass CDMA signals
-    cf["R"]["L4"] = {int(1): 1600.995 * cHzPerMhz}
```
Comment (jtec) [inline src/prx/constants.py:69]: frequency was missing

File: src/prx/constants.py [hunk 2]
```diff
@@ -66,8 +66,9 @@ def carrier_frequencies_hz():
...
     cf["R"]["L3"] = {int(1): 1202.025 * cHzPerMhz}
+    cf["R"]["L4"] = {int(1): 1600.995 * cHzPerMhz}
+    cf["R"]["L6"] = {int(1): 1248.060 * cHzPerMhz}
```
Comment (jtec) [inline src/prx/constants.py:71]: frequency was missing

File: src/prx/main.py [hunk 3]
```diff
@@ -483,7 +489,7 @@ def signal_2_carrier_frequency(row):
             flat_obs.loc[
                 mask,
                 "iono_delay_m",
-            ] = 0
+            ] = np.nan
```
Comment (jtec) [inline src/prx/main.py:492]: @plutonheaven not 100% sure here - do we want 0 or NaN for elements we cannot compute?

File: src/prx/rinex_nav/test/benchmark.py [hunk 4]
```diff
@@ -17,8 +17,6 @@ def generate_query(n_epochs=1):
     for _, constellation_df in df.groupby("constellation"):
         constellation_sats = constellation_df["sv"].unique()
         sats.extend(constellation_sats[0 : min(10, len(constellation_sats))])
-    # Only Kepler orbits supported:
```
Comment (jtec) [inline src/prx/rinex_nav/test/benchmark.py:20]: GLONASS and SBAS supported now

File: src/prx/test/benchmark.py
Comment (jtec) [inline src/prx/test/benchmark.py:1]: Convenience module for finding bottlenecks and checking the effect of changes on prx's processing speed.

File: src/prx/main.py (hunk 3 above)
Comment (plutonheaven) [reply src/prx/main.py:492]: Ok, let's put a NaN. Users will always be able to replace this by 0 if they want to.

**Review by plutonheaven — APPROVED** [review]
Nice

**Pull request merged**